ztp/kube-compare-reference/compare.sh

```diff
@@ -0,0 +1,60 @@
+#! /bin/bash
+trap cleanup EXIT
+
+function cleanup() {
+  rm -rf source_file rendered_file same_file
+}
+
+function read_dir() {
+  local dir=$1
+  local file
+
+  for file in $(ls "$dir"); do
+    if [ -d "$dir""/""$file" ]; then
+      read_dir "$dir""/""$file"
+    else
+      echo "$dir""/""$file"
+    fi
+  done
+}
+
+
+function compare_cr {
+  local rendered_dir=$1
+  local source_dir=$2
+  status=0
+
+  read_dir "$rendered_dir" |grep yaml  > rendered_file
+  read_dir "$source_dir" |grep yaml  > source_file
+
+
+  while IFS= read -r file1; do
+    while IFS= read -r file2; do
+      if [ "${file1##*/}" = "${file2##*/}" ]; then
+        diff -u "$file1" "$file2"
+        status=$(( "$status" || $? ))
```

Comment:
Inside a an arithmetic evaluation like this, you don't need '$' on your variables (But you still do on `$?` for a reason I don't fully understand).

This is equivalent:
```suggestion
        status=$(( status || $? ))
```

Reply:
Not really important, as I think this works fine as-is.